src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -595,6 +595,37 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
         tmp = nullptr;
     }
 
+    void visit_Import(const AST::Import_t &x) {
+        ASR::symbol_t *t = nullptr;
+        std::string rl_path = get_runtime_library_dir();
+        SymbolTable *st = current_scope;
+        std::vector<std::string> mods;
+        for (size_t i=0; i<x.n_names; i++) {
+            mods.push_back(x.m_names[i].m_name);
+        }
+        if (!main_module) {
+            st = st->parent;
+        }
+        for (auto &mod_sym : mods) {
+            bool ltypes;
+            t = (ASR::symbol_t*)(load_module(al, st,
+                mod_sym, x.base.base.loc, false, rl_path, ltypes,
+                [&](const std::string &msg, const Location &loc) { throw SemanticError(msg, loc); }
+                ));
```

Comment:
We visit import in symbol table visitor and store the module name in `current_scope`. But still the following snippet gives the error:
```py
import math

def f():
    i: i32
    i = math.factorial(3)
    print(i)
```
ASR:
```
semantic error: module ''math' is not imported
 --> expr3.py:1:1
  |
1 | import math
  | ^ 
```